app/assets/javascripts/discourse/app/components/topic-map/private-message-map.gjs

```diff
@@ -0,0 +1,257 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import DButton from "discourse/components/d-button";
+import { groupPath } from "discourse/lib/url";
+import dIcon from "discourse-common/helpers/d-icon";
+import { tinyAvatar } from "discourse-common/lib/avatar-utils";
+import I18n from "discourse-i18n";
+import and from "truth-helpers/helpers/and";
+
+export default class PrivateMessageMap extends Component {
+  @service site;
+  @tracked isEditing = false;
+
+  get participantsClasses() {
+    if (
+      !this.isEditing &&
+      this.site.mobileView &&
+      this.args.postAttrs.allowedGroups.length > 4
+    ) {
+      return "participants hide-names";
+    }
+    return "participants";
+  }
+
+  get canInvite() {
+    return this.args.postAttrs.canInvite;
+  }
+
+  get canRemove() {
+    return (
+      this.args.postAttrs.canRemoveAllowedUsers ||
+      this.args.postAttrs.canRemoveSelfId
+    );
+  }
+
+  get canShowControls() {
+    return this.canInvite || this.canRemove;
+  }
+
+  get actionAllowed() {
+    return this._actionMap[this._actionAllowedKey()].bind(this);
+  }
+
+  _actionAllowedKey() {
+    if (this.canInvite && this.canRemove) {
+      return "edit";
+    }
+    if (!this.canInvite && this.canRemove) {
+      return "remove";
+    }
+    return "add";
+  }
+
+  get _actionMap() {
```

Comment:
We could use the native private method syntax `#actionAllowedKey` instead of the underscore